instrumentation/github.com/Shopify/sarama/otelsarama/producer.go

```diff
@@ -227,11 +228,21 @@ func WrapAsyncProducer(saramaConfig *sarama.Config, p sarama.AsyncProducer, opts
 	return wrapped
 }
 
-func startProducerSpan(cfg config, version sarama.KafkaVersion, msg *sarama.ProducerMessage) trace.Span {
+// producerSpan is a simple wrapper around trace.Span (which can be nil) that
+// simplifies the logic in the wrapper in the case that no span is really present.
+type producerSpan struct {
```

Comment:
I'm not sure about this wapper. startProducerSpan and startProducerSpan seem paired well, but now the wrapper breaks this.

Reply:
The code without any additional type was not readable, contained more nesting, and was a lot more error-prone.